src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -411,6 +411,23 @@ internal static void ConfigureApplicationInsights(HostBuilderContext context, IL
                         o.EnableDependencyTracking = false;
                     });
                 }
+
+                if (SystemEnvironment.Instance.IsApplicationInsightsAgentEnabled())
+                {
+                    // Skip sending user generated logs to AI and QuickPulse if worker AI agent is configured, worker will send these logs to AI and Quickpulse service.
+                    builder.Services.Configure<LoggerFilterOptions>(options => options.AddFilter<ApplicationInsightsLoggerProvider>((category, logLevel) =>
+                    {
```

Comment:
There should be a way to get an `IEnvironment` inside here so you don't have to use `SystemEnvironment.Instance` -- that'd make things more testable.

I think it's like this, although I didn't confirm...

```csharp
builder.Services.AddOptions<LoggerFilterOptions>()
    .Configure<IEnvironment>((options, environment) => // environment comes from DI now
    {
        if (environment.IsApplicationInsightsAgentEnabled())
        {
            options.AddFilter<ApplicationInsightsLoggerProvider>((category, logLevel) =>
            {
                // etc...
            });
        }
    });
```